pkg/agent/controller/supportbundlecollection/support_bundle_controller.go

```diff
@@ -0,0 +1,357 @@
+	// e.g. timeout error. Make sure that watcher is not empty and log warning otherwise.
+	if reflect.TypeOf(watcher) == reflect.TypeOf(emptyWatch) {
+		klog.Warning("Failed to start watch for SupportBundleCollections, please ensure antrea service is reachable for the agent")
+		return
+	}
+
+	klog.Info("Started watch for SupportBundleCollections")
+	eventCount := 0
+	defer func() {
+		klog.Infof("Stopped watch for SupportBundleCollections, total items received: %d", eventCount)
+		watcher.Stop()
+	}()
+
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return
+			}
+			switch event.Type {
+			case watch.Bookmark:
+				klog.Infof("Received Bookmark event")
+			case watch.Added:
+				c.enqueueSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.Infof("Enqueue added SupportBundleCollections (%#v)", event.Object)
+			case watch.Modified:
+				klog.V(2).Infof("Received updated SupportBundleCollections (%#v)", event.Object)
+			case watch.Deleted:
+				c.enqueueSupportBundleCollection(event.Object.(*cpv1b2.SupportBundleCollection))
+				klog.Infof("Enqueue removed SupportBundleCollections (%#v)", event.Object)
+			default:
+				klog.Errorf("Unknown event: %v", event)
+				return
+			}
+			eventCount++
+		}
+
+	}
+}
+
+func (c *SupportBundleController) enqueueSupportBundleCollection(supportbundle *cpv1b2.SupportBundleCollection) {
+	c.queue.Add(supportbundle.Name)
+}
+
+func (c *SupportBundleController) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.InfoS("Starting", "controllerName", controllerName)
+	defer klog.InfoS("Shutting down", "controllerName", controllerName)
+
+	go wait.NonSlidingUntil(c.watchSupportBundleCollections, 5*time.Second, stopCh)
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+	<-stopCh
+}
+
+func (c *SupportBundleController) worker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *SupportBundleController) processNextWorkItem() bool {
+	obj, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(obj)
+
+	if key, ok := obj.(string); !ok {
+		// As the item in the workqueue is actually invalid, we call Forget here else we'd
+		// go into a loop of attempting to process a work item that is invalid.
+		// This should not happen.
+		klog.Errorf("Expected string in work queue but got %#v", obj)
+		return true
+	} else if err := c.syncSupportBundleCollection(key); err == nil {
+		klog.Infof("Successfully collected support bundle %s", key)
+	} else {
+		// Skip retrying as the time may not meet the requirements for SupportBundle.
+		klog.Errorf("Error syncing SupportBundleCollection %s, skip retrying. Error: %v", key, err)
+	}
+	return true
+}
+
+func (c *SupportBundleController) syncSupportBundleCollection(key string) error {
+	klog.Infof("Sync support bundle collection %s", key)
+	antreaClient, err := c.antreaClientGetter.GetAntreaClient()
```

Comment:
Why not use `antreaClient` as a property in Controller instead of antreaClientGetter?

Reply:
All the other modules of agent side use this pattern, so I think we should follow the same.